ArduCopter/mode_guided.cpp

```diff
@@ -1159,4 +1199,29 @@ uint32_t ModeGuided::get_timeout_ms() const
     return MAX(copter.g2.guided_timeout, 0.1) * 1000;
 }
 
+/********************************************************************************/
+// Pause and continue guided mode
+/********************************************************************************/
+
+// pause guide mode
+bool ModeGuided::pause()
+{
+    _paused = true;
+    return true;
+}
+
+// resume guided mode
+bool ModeGuided::resume()
+{
+    _paused = false;
+    return true;
+}
+
+// exit guided mode
+void ModeGuided::exit()
+{
+    // clear pause state when exiting guided mode
+    _paused = false;
```

Comment:
Instead of initialising this variable on exit, let's initialise it on entry because that's where we do all the other initialisation and I don't think we need it in both places do we?